PythonVisualizations/orderedArray.py

```diff
@@ -0,0 +1,491 @@
+        self.buttons = self.makeButtons()
+        
+        # Fill in initial array values with random integers
+        # The display items representing these array cells are created later
+        a = [random.randrange(90) for i in range(size-1)]
+        a.sort()
+        
+        # Append and draw them to the list and draw them
+        for i in a:
+            self.list.append(drawable(i))
+
+        self.display()        
+
+    def __str__(self):
+        return str(self.list)
+
+    # ARRAY FUNCTIONALITY
+
+    def createIndex(         # Create an index arrow to point at an indexed
+            self, index,     # cell
+            name=None,       # with an optional name label
+            level=1,         # at a particular level away from the cells
+            color=VARIABLE_COLOR): # (negative are below)
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        level_spacing = VARIABLE_FONT[1] 
+        x = cell_center[0]
+        if level > 0:
+            y0 = cell_coords[1] - CELL_SIZE * 3 // 5 - level * level_spacing
+            y1 = cell_coords[1] - CELL_SIZE * 3 // 10
+        else:
+            y0 = cell_coords[3] + CELL_SIZE * 3 // 5 - level * level_spacing
+            y1 = cell_coords[3] + CELL_SIZE * 3 // 10
+        arrow = self.canvas.create_line(
+            x, y0, x, y1, arrow="last", fill=color)
+        if name:
+            label = self.canvas.create_text(
+                x + 2, y0, text=name, anchor=SW if level > 0 else NW,
+                font=VARIABLE_FONT, fill=color)
+        return (arrow, label) if name else (arrow, )
+
+    insertCode = """
+def insert(self, item):
+   self.__a[self.__nItems] = item
+   self.__nItems += 1
+"""
+    insertCodeSnippets = {
+        'item_assignment': ('2.3', '2.end'),
+        'nitem_increment': ('3.3', '3.end'),
+    }
+    
+    def insert(self, val):
+<<<<<<< Updated upstream
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+        j = self.find(val)  # Find where item should go
+
+        self.list.append(drawable(None))
+        
+        indexK = self.createIndex(len(self.list), 'k', level=-1) # create "k" arrow
+        callEnviron |= set(indexK)  
+        
+        for k in range(len(self.list) - 1, j, -1):  # Move bigger items right
+            
+            self.wait(1)
+            for item in indexK:
+                self.canvas.move(item, -CELL_SIZE, 0)  # Move "k" arrow  
+                
+            self.list[k].val = self.list[k - 1].val 
+            self.assignElement(k - 1, k, callEnviron)    
+                    
+=======
+        self.cleanUp()
+        j = self.find(val)  # Find where item should go
+
+        self.list.append(drawable(None))
+
+        for k in range(len(self.list) - 1, j, -1):  # Move bigger items right
+            self.wait(1)
+            self.list[k].val = self.list[k - 1].val
+            self.assignElement(k - 1, k)
+
+>>>>>>> Stashed changes
```

Comment:
You must reconcile any merge conflicts, like the one here, before committing your code.  You can find them by searching for `====`.